### PR TITLE
building-blocks documentation updates

### PR DESCRIPTION
**Pull request by slyness**
Some adjustments that I thought would help the building blocks documentation in a few places.

### CODE REVIEW

**Comment by luckymike** [comment]
:+1:

File: docs/building-blocks.md [hunk 1]
```diff
@@ -215,13 +215,13 @@ which will result in a resource name: `FoobarInstance`
...
-prevent surprise for users. When `dynamic!` is called and provided a block, that
-block is evaluated within the context returned from the requested dynamic.
+A dynamic can make multiple modifications to a template when inserted. For
+example, addition of parameters, resources, and/or outputs. It is important
+to use the values returned from the dynamic block to prevent surprises.
+When the `dynamic!` is called and provided a block, the block is evaluated
+within the context returned from the `dynamic!`.
 
-For example, this is a poor implementation of a dynamic:
+For example, this is a improper implementation of a dynamic:
```
Comment (chrisroberts) [inline docs/building-blocks.md]: > this is _an_ improper

File: docs/building-blocks.md [hunk 2]
```diff
@@ -258,7 +258,7 @@ SparkleFormation.dynamic(:bad_dynamic) do |_name, _config|
 end
 ~~~
 
-This ensures the instance resource is the context returned, and provided blocks
+This will ensure the instance resource is the context returned. The blocks produced
```
Comment (chrisroberts) [inline docs/building-blocks.md]: I think that "provided" works better here than "produced"

File: docs/building-blocks.md [hunk 3]
```diff
@@ -297,7 +297,7 @@ end
 SfnRegistry.register(:instance_size_default){ 'm3.medium' }
 ~~~
 
-With the value registered, it can then be referenced:
+With the value registered holding an array of values it can then be referenced:
```
Comment (chrisroberts) [inline docs/building-blocks.md]: Can you toss a comma after `values`:

> holding an array of values,

**Comment by chrisroberts** [comment]
@slyness Can you update those couple items and i'll merge this in. Thanks so much man! :metal:

**Comment by slyness** [comment]
@chrisroberts updated!

File: docs/building-blocks.md [hunk 4]
```diff
@@ -297,7 +297,7 @@ end
 SfnRegistry.register(:instance_size_default){ 'm3.medium' }
 ~~~
 
-With the value registered, it can then be referenced:
+With the value registered, holding an array of values, it can then be referenced:
```
Comment (luckymike) [inline docs/building-blocks.md]: This reads oddly to me. Maybe: "With the array registered as `instance_size`, we can now reference it:"

**Comment by luckymike** [comment]
Can you reopen against Develop?

**Comment by slyness** [comment]
Originally noticed that most documentation PRs hadn't recently gone through via develop. But see #80 for such a PR.

**Comment by luckymike** [comment]
Closing because #80

**Pull request closed without merging**